### PR TITLE
lua: add stats

### PR DESCRIPTION
**Pull request by surki**
Commit Message: lua: add stats
Additional Description: Currently stats are not available in lua filter, which makes it hard to
track/alert for the script errors. This change adds error stat.
Risk Level: Low
Testing: Manual and Unit tests
Docs Changes: Done
Release Notes: Done
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by kyessenov** [comment]
(re-assigning from @markdroth to @mattklein123 for API review)

File: changelogs/current.yaml [hunk 1]
```diff
@@ -98,5 +98,8 @@ new_features:
 - area: access_log
   change: |
     added support for number values in substitution format string in json_format.
+- area: lua
+  change: |
+    added stats for lua filter.
```
Comment (mattklein123) [inline changelogs/current.yaml:103]: please ref link to the new section you add in lua_filter.rst

File: api/envoy/extensions/filters/http/lua/v3/lua.proto [hunk 2]
```diff
@@ -58,6 +58,23 @@ message Lua {
   // The default Lua code that Envoy will execute. If no per route config is provided
   // for the request, this Lua code will be applied.
   config.core.v3.DataSource default_source_code = 3;
+
+  // Optional additional prefix to use when emitting statistics. This allows to distinguish
+  // emitted statistics between configured ``lua`` filters in an HTTP filter chain. For example:
```
Comment (mattklein123) [inline api/envoy/extensions/filters/http/lua/v3/lua.proto:63]: Specify what the default is which I think is 'lua' ? Or maybe the default is empty? Unclear.

**Review by mattklein123 — CHANGES_REQUESTED** [review]
LGTM with small comments, thanks.

/wait

File: changelogs/current.yaml (hunk 1 above)
Comment (surki) [reply changelogs/current.yaml]: Done

File: api/envoy/extensions/filters/http/lua/v3/lua.proto (hunk 2 above)
Comment (surki) [reply api/envoy/extensions/filters/http/lua/v3/lua.proto]: The default is `lua`, updated the doc to be more explicit

File: docs/root/configuration/http/http_filters/lua_filter.rst [hunk 3]
```diff
@@ -155,6 +155,19 @@ Lua script as follows:
             response_handle:logInfo("Goodbye.")
           end
 
+Statistics
+----------
+.. _config_http_filters_lua_stats:
+
+The lua filter outputs statistics in the *.lua.* namespace by default. When
```
[the remark this answers is not in the document]
Comment (surki) [reply docs/root/configuration/http/http_filters/lua_filter.rst:162]: Updated

**Review by mattklein123 — APPROVED** [review]
Thanks!

**Pull request merged**